src/apack/zcl_abapgit_apack_migration.clas.abap

```diff
@@ -0,0 +1,217 @@
+CLASS zcl_abapgit_apack_migration DEFINITION
+  PUBLIC
+  FINAL
+  CREATE PRIVATE.
+
+  PUBLIC SECTION.
+    CLASS-METHODS: run RAISING zcx_abapgit_exception.
+    METHODS: perform_migration RAISING zcx_abapgit_exception.
+  PROTECTED SECTION.
+  PRIVATE SECTION.
+
+    CONSTANTS: c_interface_name TYPE seoclsname VALUE 'ZIF_APACK_MANIFEST' ##NO_TEXT.
+
+    METHODS:
+      interface_exists RETURNING VALUE(rv_interface_exists) TYPE abap_bool,
+      create_interface RAISING zcx_abapgit_exception,
+      add_interface_source_classic IMPORTING is_clskey TYPE seoclskey
+                                   RAISING   zcx_abapgit_exception,
+      add_interface_source IMPORTING is_clskey TYPE seoclskey
+                           RAISING   zcx_abapgit_exception,
+      get_interface_source RETURNING VALUE(rt_source) TYPE zif_abapgit_definitions=>ty_string_tt.
+ENDCLASS.
+
+
+
+CLASS zcl_abapgit_apack_migration IMPLEMENTATION.
+
+
+  METHOD add_interface_source_classic.
+    DATA: lo_source      TYPE REF TO cl_oo_source,
+          lt_source_code TYPE zif_abapgit_definitions=>ty_string_tt.
+
+    CREATE OBJECT lo_source
+      EXPORTING
+        clskey             = is_clskey
+      EXCEPTIONS
+        class_not_existing = 1
+        OTHERS             = 2.
+    IF sy-subrc <> 0.
+      zcx_abapgit_exception=>raise( 'error from CL_OO_SOURCE' ) ##NO_TEXT.
+    ENDIF.
+
+    TRY.
+        lo_source->access_permission( seok_access_modify ).
+        lt_source_code = get_interface_source( ).
+        lo_source->set_source( lt_source_code ).
+        lo_source->save( ).
+        lo_source->access_permission( seok_access_free ).
+      CATCH cx_oo_access_permission.
+        zcx_abapgit_exception=>raise( 'permission error' ) ##NO_TEXT.
```

Comment:
Are you sure the user will know what does the error text "permission error" means?

Reply:
It's the same error message as for any other class/interface creation within abapGit - its actually copied. So, if it was OK until now, I would answer: Yes, I'm sure. :)

Reply:
Honestly, I am not happy with error reporting in abapgit (sure, you see no PRs from me, only because I am too lazy) ... I don't think it is ok to be so quite and fuzzy. However, now, we have a great opportunity to do the error reporting better.

Reply:
Well, it's only a small additional functionality using the same error handling as anywhere else. I'd rather keep different concerns separated. So if error handling is improved it can be tackled in a different PR.